src/dispatch/builder.rs

```diff
@@ -93,10 +93,10 @@ use crate::{
 pub struct DispatcherBuilder<'a, 'b> {
     current_id: usize,
     map: HashMap<String, SystemId>,
-    stages_builder: StagesBuilder<'a>,
+    pub(crate) stages_builder: StagesBuilder<'a>,
     thread_local: ThreadLocal<'b>,
     #[cfg(feature = "parallel")]
-    thread_pool: Option<::std::sync::Arc<::rayon::ThreadPool>>,
+    thread_pool: ::std::sync::Arc<::std::sync::RwLock<ThreadPoolWrapper>>,
```

Comment:
There is an `Arc` inside `ThreadPoolWrapper`. I assume that the outer `Arc` is still needed, right?

Reply:
The outer Arc is necessary because the wrapper is shared across the dispatcher and its internal batches.
Doing so allow change the ThreadPool to the main dispatcher and all its batches.

The internal Arc is needed because the ThreadPool could come from someone that is not using the Wrapper, for example the Amethyst thread pool.